app/src/main/java/woowacourse/shopping/ui/payment/PaymentViewModel.kt

```
@@ -0,0 +1,129 @@
+                }
+            payment.copy(selectedCoupon = selectedCoupon)
+        }
+    }
+
+    fun completePayment(onSuccess: () -> Unit) {
+        viewModelScope.launch {
+            try {
+                orderRepository.createOrder(selectedCartItemIds)
+                _uiEvent.emit(UiEvent.ShowMessage("주문이 완료되었습니다."))
+                onSuccess()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)